centreon/www/front_src/src/Resources/Listing/columns/index.tsx

```diff
@@ -265,28 +290,57 @@ export const getColumns = ({
       getRenderComponentOnRowUpdateCondition: T,
       id: 'checks',
       label: t(labelCheck),
-      rowMemoProps: ['passive_checks', 'active_checks'],
+      rowMemoProps: ['has_passive_checks_enabled', 'has_active_checks_enabled'],
       shortLabel: 'C',
       type: ColumnType.component
     }
   ];
 
   if (equals(visualization, Visualization.Service)) {
-    const columnsForVisualizationByService = columns
-      .map((column) =>
-        equals(column.label, t(labelResource))
-          ? { ...column, label: t(labelService) }
-          : column
-      )
-      .map((column) =>
-        equals(column.label, t(labelParent))
-          ? { ...column, label: t(labelHost) }
-          : column
-      );
+    const changeResourceLabel = (column: Column): Column =>
+      equals(column.label, labelResource)
+        ? { ...column, label: t(labelService) }
+        : column;
+
+    const changeParentLabel = (column: Column): Column =>
+      equals(column.label, labelParent)
+        ? { ...column, label: t(labelHost) }
+        : column;
+
+    const columnsForVisualizationByService = pipe(
+      map(changeResourceLabel),
+      map(changeParentLabel)
+    )(columns);
 
     return columnsForVisualizationByService;
   }
 
+  if (equals(visualization, Visualization.Host)) {
+    const subItemColumn = {
+      Component: SubItem,
+      displaySubItemsCaret: true,
+      id: 'services',
+      label: t(labelServices),
+      type: ColumnType.component,
+      width: 'max-content'
+    };
+
+    const changeResourceLabel = (column: Column): Column =>
+      equals(column.label, labelResource)
+        ? { ...column, label: t(labelHost) }
+        : column;
+
+    const columnsForVisualizationByHost = pipe(
+      filter(complement(propEq('parent_resource', 'id'))),
+      filter(complement(propEq('status', 'id'))),
+      filter(complement(propEq('parent_alias', 'id'))),
```

Comment:
Is the `filter(complement(propEq(...))` behaves like `reject(propEq(...))`?
If so, please use the reject in order to avoid to filter on a negation. It is like: I filter on a property that is not. I would something like: I reject property that is

Reply:
exactly, reject is more appropriate for this case